src/TextToTalk/TextToTalk.cs

```diff
@@ -391,6 +406,38 @@ private void BackendSay(SayRequest request)
             this.backendManager.Say(request);
         }
 
+        private unsafe string GetSpeakerRace(GameObject? speaker)
+        {
+            var race = this.data.GetExcelSheet<Race>();
+            if (race is null || speaker is null || speaker.Address == nint.Zero)
+            {
+                return "Unknown";
+            }
+
+            var charaStruct = (FFXIVClientStructs.FFXIV.Client.Game.Character.Character*)speaker.Address;
+            var speakerRace = charaStruct->DrawData.CustomizeData.Race;
+            var row = race.GetRow(speakerRace);
+
+            if (row is null)
+            {
+                return "Unknown";
+            }
+
+            return row.Masculine;
+        }
+
+        private unsafe BodyType GetSpeakerBodyType(GameObject? speaker)
```

Comment:
Super non-critical nit: since this doesn't use any instance state, it could be moved to GameObjectExtensions as an extension method (or just to a util file), to declutter the main file a bit.